### PR TITLE
Changed way of filling embedded tables with data

### PR DESCRIPTION
**Pull request by kkursor**
As we discussed here https://github.com/pgmodeler/pgmodeler/issues/1334
I did this change and it works good.
I have a model which has lots of pre-defined data (4 mb of CSV catalogs) and I prefer to store them in model than to store them separatedly.
It's easier to provide pre-filled model than model and supplementary data which somebody has to insert with possibility to do something wrong.

### CODE REVIEW

**Comment by rkhaotix** [comment]
This ideia sounds good! But what if we could have the `COPY` / `INSERT` commands generated by option? I think that instead of replacing the `INSERT` you can add an option to use the data provided in that dialog in for of `COPY` or `INSERT`. 

Just a question... Did you tested this feature in the validation widget? How was the behavior?

**Comment by kkursor** [comment]
Yes, checkbox at export tab is probably a better solution, because INSERT query is the only way to insert rows with calculable values (eg NOW() or any other built-in or third-party function).

Model completes validation without problems. 

Or which widget do you mean?

**Comment by kkursor** [comment]
Just found that export directly to database does not work.
Need to be tested more thoroughly.
I'll be back (c)

File: libpgconnector/src/connection.cpp [hunk 1]
```diff
@@ -437,7 +437,61 @@ void Connection::executeDDLCommand(const QString &sql)
...
+            if ( 1 ==  PQputCopyEnd( connection, NULL ) )
+            {
+                sql_res = PQgetResult(connection);
+                if (PQresultStatus(sql_res) != PGRES_COMMAND_OK) 
+                {
+                    QString field = QString(PQresultErrorField(sql_res, PG_DIAG_SQLSTATE));
+                
+                    PQclear(sql_res);
+                    
+                    throw Exception(Exception::getErrorMessage(ErrorCode::SQLCommandNotExecuted)
```
Comment (rkhaotix) [inline libpgconnector/src/connection.cpp]: Is there the need to send this error from here? Right below (lines 504-515) the process of raise an exception and destroy the sql_res is performed.

Comment (kkursor) [reply libpgconnector/src/connection.cpp]: I agree. Lines 468-477 are excessive

File: libpgmodeler/src/table.cpp [hunk 2]
```diff
@@ -1998,16 +2002,18 @@ QString Table::getInitialDataCommands(void)
...
+		commands.append("\\.");
 
-		return(commands.join('\n'));
+		return(commands.join('\n') + Attributes::DdlEndToken );
 	}
 
 	return(QString());
 }
 
 QString Table::createInsertCommand(const QStringList &col_names, const QStringList &values)
```
Comment (rkhaotix) [inline libpgmodeler/src/table.cpp]: Maybe this method here can be parameterized in such way to accept a boolean param in which the user (dev) decides if COPY or INSERT need to be generated?

Comment (kkursor) [reply libpgmodeler/src/table.cpp]: I haven't digged UI code good enough to answer and to find out how interaction between UI and codegenerator occurs. 
Probably we should change signature of this method from `QString Table::getInitialDataCommands(void)` to `QString Table::getInitialDataCommands( bool use_copy_syntax = false )` to provide compatibility for models that relied on INSERT syntax.

**Comment by kkursor** [comment]
Look at this. (^^,)

**Comment by rkhaotix** [comment]
Making the copy generation global (for all instances of table) may be not desirable by the user!? I thought to use it optionally per table, which means, the user would control which tables generate COPY or INSERT as initial data. Maybe I didn't make it clear... sorry! :(

**Comment by rkhaotix** [comment]
Yes, you're right, it can be hard to keep both options... well, let's leave it global for now. We need o be sure that nothing is broken after this change. Code generation and export is a crucial part o the software so let's test it exhaustedly

**Comment by kkursor** [comment]
Ok. I'm actively developing database right now and using export feature very often.
If I found something wrong, I'll let you know.

**Comment by rkhaotix** [comment]
Thanks. 
I'm working on 0.9.2-beta2 and if possible try to make a PR against the branch `0.9.2-beta2` to see if it can be merged without problems. If so, I can review, test and release this feature still on that version. :)

**Comment by kkursor** [comment]
Fixed conflicts.

**Comment by kkursor** [comment]
It seems that it's not easy to merge. Have you changed interface of `Table` class?

**Comment by kkursor** [comment]
Merged all conflicts, tested, it seems to work.

**Comment by rkhaotix** [comment]
> It seems that it's not easy to merge. Have you changed interface of `Table` class?

Yes. I changed a lot of things...  Due to the introduction of foreign tables I had to create an intermediate class that holds common attributes between tables and foreign tables. 

If you perfer, you can keep working on develop branch that I'll applying your changes manually and testing them little by little and doing the needed fixes.

**Comment by kkursor** [comment]
I've applied my patches to 0.9.2-beta2

**Pull request closed without merging**